presto-main/src/main/java/io/prestosql/execution/resourcegroups/InternalResourceGroupManager.java

```diff
@@ -92,17 +92,19 @@ public InternalResourceGroupManager(LegacyResourceGroupConfigurationManager lega
     }
 
     @Override
-    public ResourceGroupInfo getResourceGroupInfo(ResourceGroupId id)
+    public Optional<ResourceGroupInfo> tryGetResourceGroupInfo(ResourceGroupId id)
```

Comment:
I would not rename the name. It is obvious that it may not return value, so there is not need for `try` prefix in method name.

Reply:
I know that you wouldn't :)

Reply:
BTW using Optional for this purpose is as good as throwing a (checked-)exception.
Why don't we do that? Let's talk.